lib/events/filelog.go

```diff
@@ -136,12 +154,28 @@ func (l *FileLog) EmitAuditEvent(ctx context.Context, event AuditEvent) error {
 
 // EmitAuditEventLegacy adds a new event to the log. Part of auth.IFileLog interface.
 func (l *FileLog) EmitAuditEventLegacy(event Event, fields EventFields) error {
+	l.rw.RLock()
```

Comment:
So, there's still I/O being done under lock. In particular, `fmt.Fprintln(l.file, string(line))` call is what I'm worried about (and similar one in another function).

Previously, we would just rotate the file under lock which would happen infrequently but now we're basically writing each event to the file under lock, unless I'm missing anything? I know there's buffering etc. but still, feels a bit unideal. Is there a way we can avoid writing to the file under lock?

Reply:
@r0mant File is being written under *read lock* now, so it won't block other operations, except for the rotation operation, which we want this to block because we don't want to close the file while it is in use.  If blocking other operations on I/O is not your concern, could you elaborate?

Reply:
Ah yes you're correct. Should be all good then.